internal/io/http/httpserver/data_server.go

```diff
@@ -72,6 +88,97 @@ func registerInit() error {
 	return nil
 }
 
+func UnRegisterWebSocketEndpoint(endpoint string) error {
+	lock.Lock()
+	defer lock.Unlock()
+	refCount--
+	wsEndpointRefCnt[endpoint] = wsEndpointRefCnt[endpoint] - 1
+	if wsEndpointRefCnt[endpoint] < 1 {
+		wsEndpointCancel[endpoint]()
+		delete(wsEndpointCancel, endpoint)
+		delete(wsEndpointRefCnt, endpoint)
+	}
+	if refCount == 0 {
+		shutdown()
+	}
+	return nil
+}
+
+func recvProcess(ctx api.StreamContext, c *websocket.Conn, endpoint string) {
+	topic := fmt.Sprintf("recv/%s/%s", WebsocketTopicPrefix, endpoint)
+	pubsub.CreatePub(topic)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+		}
+		msgType, data, err := c.ReadMessage()
+		if err != nil {
+			if strings.Contains(err.Error(), "close") {
+				conf.Log.Infof("websocket endpoint %s connection get closed", endpoint)
+				break
+			}
+			conf.Log.Errorf("websocket endpoint %s recv error %s", endpoint, err)
+			continue
+		}
+		if msgType == websocket.TextMessage {
```

Comment:
What about other msgType? Should mention it in the doc. And the format is required to be json map, also need to mention it.